core/src/main/java/com/oath/oak/OakRValueBufferImpl.java

```diff
@@ -50,30 +47,24 @@
         this.valueOperator = valueOperator;
         this.memoryManager = memoryManager;
         this.internalOakMap = internalOakMap;
-    }
-
-    private Slice getValueSlice() {
-        int[] valueArray = UnsafeUtils.longToInts(valueReference);
-        return memoryManager.getSliceFromBlockID(valueArray[BLOCK_ID_LENGTH_ARRAY_INDEX] >>> VALUE_BLOCK_SHIFT,
-                valueArray[POSITION_ARRAY_INDEX], valueArray[BLOCK_ID_LENGTH_ARRAY_INDEX] & VALUE_LENGTH_MASK);
+        this.s = EntrySet.buildValueSlice(valueReference, version, memoryManager);
```

Comment:
Slice is only created once (here).
This means that this modification does not support moveable values.
See PR https://github.com/yahoo/Oak/pull/86 for more information.

Reply:
this was fixed before